.github/workflows/pr_build.yaml

```
@@ -76,6 +76,25 @@ jobs:
       - name: Run unit tests
         run: ./.github/workflows/scripts/run_unit_tests.sh
 
+  unit-test-race-detector:
+    name: Run unit tests with race detector (linux)
```

The current incarnation definitely looks different than the rest and stands out (in a not great way :) ). How about just something simple and consistent with the other job names, like:
```suggestion
    name: unit-test (linux with race detection)
```